src/event-manager/event-channel-client.ts

```diff
@@ -0,0 +1,129 @@
+import Ain from '../ain';
+import { WebSocket } from 'ws';
+import {
+  EventChannelMessageTypes,
+  EventChannelMessage,
+  BlockchainEventTypes,
+  EventChannelConnectionOption,
+} from '../types';
+import EventFilter from './event-filter';
+import EventCallbackManager from './event-callback-manager';
+
+export default class EventChannelClient {
+  private readonly _ain: Ain;
+  private readonly _eventCallbackManager: EventCallbackManager;
+  private _wsClient?: WebSocket;
+  private _endpointUrl?: string;
+  private _isConnected: boolean;
+
+  constructor(ain: Ain, eventCallbackManager: EventCallbackManager) {
+    this._ain = ain;
+    this._eventCallbackManager = eventCallbackManager;
+    this._wsClient = undefined;
+    this._endpointUrl = undefined;
+    this._isConnected = false;
+  }
+
+  get isConnected(): boolean {
+    return this._isConnected;
+  }
+
+  connect(connectionOption: EventChannelConnectionOption) {
+    return new Promise(async (resolve, reject) => {
+      const eventHandlerNetworkInfo = await this._ain.net.getEventHandlerNetworkInfo();
+      const url = eventHandlerNetworkInfo.url;
+      if (!url) {
+        reject(new Error(`Can't get url from eventHandlerNetworkInfo ` +
+            `(${JSON.stringify(eventHandlerNetworkInfo, null, 2)}`));
+      }
+      this._endpointUrl = url;
+      this._wsClient = new WebSocket(url, [], { handshakeTimeout: connectionOption.handshakeTimeout || 30000 });
```

Comment:
question: when the handshake timeout triggers, is the 'error' callback called?

Reply:
Yes! I tested it.